cli/stork-publisher-agent/runner.go

```diff
@@ -76,23 +76,23 @@ func (r *PublisherAgentRunner[T]) UpdateBrokerConnections() {
 
 	// add or update desired connections
 	for brokerUrl, newAssetIdMap := range newBrokerMap {
-		_, exists := r.brokerMap[brokerUrl]
-		if exists {
+		outgoingConnection, outgoingConnectionExists := r.outgoingConnectionsByBroker[brokerUrl]
```

Comment:
the old logic here was kind of weird - we really only want to update connections which still exist, regardless of whether we have an existing entry in the broker asset map. 

We now only look at the connection map when deciding whether to create a new connection, and the asset map is basically a configuration for each connection.